pkg/controller/common.go

```diff
@@ -1130,3 +1129,36 @@ func WatchObject(mgr manager.Manager, objKind source.Kind) error {
 	})
 	return err
 }
+
+// WatchCRD creates a noop controller to set up a watch and an informer for CRD objects in controller-runtime manager
+func CacheCRD(mgr manager.Manager) error {
+	c, err := controller.New("CRD-controller", mgr, controller.Options{
+		Reconciler: &noOpReconciler{},
+	})
+	if err != nil {
+		return err
+	}
+
+	c.GetLogger().Info("Starting to watch CRDs")
+
+	// Watch for CRUD events on CRD objects
+	err = c.Watch(&source.Kind{Type: &apiext.CustomResourceDefinition{}}, &handler.EnqueueRequestForObject{}, predicate.Funcs{
+		CreateFunc: func(_ event.CreateEvent) bool {
+			return false
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
```

Comment:
We don't need to return true in the predicate here because we have another informer for CRDs and this is a noop controller. I am not sure why `ChacheCRD` and `CacheNamespace` have been refactored into two different methods when they could have been one `CacheObject` method taking the object type as a parameter.

Reply:
If the predicate isn't needed, I can change it to only one `CacheObject` method.

Reply:
> CacheObject method taking the object type as a parameter.
This wouldn't do for CRD case as it requires a manual predicate. 

We either need to create `CacheObject(mgr manager.Manager, obj runtime.Object, options ...controller.WatchOption)` to allow for option to manually configure predicates (like we do for workflow objects) https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/a2fde828b8a4fd9e3de24543861583b5a742a048/pkg/workflow/workflow.go#L84-L120

Or we need them as separate functions.

I personally think using `WatchOption` would give the code more versatility

Reply:
We have an informer that triggers a defined update function for the CRD. Why do we need a predicate that returns true for a noop controller here?

Reply:
We don't have an informer that triggers anything for the CRD. We need a predicate to detect if `AzVolume` or `AzVolumeAttachment` CRD is requested for deletion. 

As I commented above, I believe we should use atomic value to denote if the driver is being uninstalled or not, like we do for `isRecoveryComplete`, instead of listing all the CRDs every time `isDriverUninstall()` is called.

Reply:
Here is the informer: https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/1570/files#diff-969dc63b611428d6a3320f889ab448a42f1288c3a211533f19dc2a7bf526ef75R1250

Reply:
Oh, I just realize that we can have predicate but it only returns true. If that, I created two functions because their predicates are different, and defining predicates as function parameter in `azureisk_v2` makes the code difficult to read that you commented earlier in the PR.

Reply:
That informer lives in `crdProvsioner` not in the controllers. We don't want the `crdProvisioner` and controller to have shared memory.

Reply:
I believe we should use the informer instead of the predicate to achieve that functionality as it is currently done. All predicates for caching should return false for a noop controller. And predacities should not be initializing any values as their purpose is to decide if an event should be triggered for reconciliation or not.

Reply:
We should create an informer in the controller for the CRDs as well if that is the case.

Reply:
Internally, it's exactly the same thing unless we use a queue, and worker goroutines to dequeue objects and run reconciliation on them, which is the standard way to write controllers. Handling things directly in event handler vs. handling in controller runtime predicate func is essentially going to be no different.

Reply:
I understand that such usage of predicate func goes against its intended use case but I also am not a big fan of adding bunch of shared informers when we can leverage controller-runtime pkg to centrally manage cache.